sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/GenericInheritanceShouldNotBeRecursive.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace Tests.SingleClassRecursion
+{
+    // some base classes
+    class CA<T> { }
+    class CB<T> { }
+    class CC<T1, T2> { }
+
+    class C0<T> : CA<C0<T>> { } // Compliant
+
+    class C1<T> : CA<CB<C1<T>>> { } // Compliant
+
+    class C2<T> : CA<C2<CB<T>>> { } // Noncompliant
+
+    class C3<T> : CA<C3<C3<T>>> { } // Noncompliant
+
+    class C4<T> : CA<C4<CA<T>>> { } // Noncompliant
+
+    class C5<T> : CC<C5<CA<T>>, CB<T>> { } // Noncompliant

```

Comment:
Please also add a test where only one of the 2 generic argument is causing the inifinite looping.